indexer/feature_meta.hpp

```diff
@@ -84,56 +110,36 @@ namespace feature
         uint8_t const mark = (&e == &(*m_metadata.crbegin()) ? 0x80 : 0);
         uint8_t elem[2] = {static_cast<uint8_t>(e.first | mark),
                            static_cast<uint8_t>(min(e.second.size(), (size_t)kMaxStringLength))};
-        ar.Write(elem, sizeof(elem));
-        ar.Write(e.second.data(), elem[1]);
+        writer.Write(elem, sizeof(elem));
+        writer.Write(e.second.data(), elem[1]);
       }
     }
 
-    template <class ArchiveT> void DeserializeFromMWM(ArchiveT & ar)
+    template <class TSource> void DeserializeFromMWM(TSource & src)
     {
       uint8_t header[2] = {0};
       char buffer[kMaxStringLength] = {0};
       do
       {
-        ar.Read(header, sizeof(header));
-        ar.Read(buffer, header[1]);
-        m_metadata[ToType(header[0] & 0x7F)].assign(buffer, header[1]);
+        src.Read(header, sizeof(header));
+        src.Read(buffer, header[1]);
+        m_metadata[header[0] & 0x7F].assign(buffer, header[1]);
       } while (!(header[0] & 0x80));
     }
 
-    template <class ArchiveT> void Serialize(ArchiveT & ar) const
-    {
-      uint8_t const sz = m_metadata.size();
-      WriteToSink(ar, sz);
-      for (auto const & it : m_metadata)
-      {
-        WriteToSink(ar, static_cast<uint8_t>(it.first));
-        utils::WriteString(ar, it.second);
-      }
-    }
-
-    template <class ArchiveT> void Deserialize(ArchiveT & ar)
-    {
-      uint8_t const sz = ReadPrimitiveFromSource<uint8_t>(ar);
-      ASSERT_LESS_OR_EQUAL(sz, FMD_COUNT, ());
+  private:
+    enum { kMaxStringLength = 255 };
+  };
 
-      for (size_t i = 0; i < sz; ++i)
-      {
-        EType const key = ToType(ReadPrimitiveFromSource<uint8_t>(ar));
-        string value;
-        utils::ReadString(ar, value);
-        m_metadata.insert(make_pair(key, value));
-      }
-    }
+  class AddressData : public MetadataBase
+  {
+  public:
+    enum Type { PLACE, STREET, POSTCODE };
```

Comment:
According to our style guide these enum values should be named as TYPE_PLACE, TYPE_STREET, TYPE_POSTCODE